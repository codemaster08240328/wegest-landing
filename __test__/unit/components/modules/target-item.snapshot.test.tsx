import React from 'react'
import renderer from 'react-test-renderer'
import TargetItem from '@/components/modules/target-item'

describe('<TargetItem> snapshot', () => {
  it('renders correctly when there are no props', () => {
    expect(renderer.create(<TargetItem />).toJSON()).toMatchSnapshot()
  })

  it('renders correctly when there title is passed', () => {
    expect(
      renderer.create(<TargetItem title='Shave' />).toJSON(),
    ).toMatchSnapshot()
  })

  it('renders correctly when there stats are passed', () => {
    expect(renderer.create(<TargetItem total={3} />).toJSON()).toMatchSnapshot()
  })

  it('renders correctly when there are title and stats', () => {
    expect(
      renderer
        .create(<TargetItem title='Pedicure' done={1} total={4} />)
        .toJSON(),
    ).toMatchSnapshot()
  })
})
