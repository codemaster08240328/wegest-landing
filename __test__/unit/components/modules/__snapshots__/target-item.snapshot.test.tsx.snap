// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TargetItem> snapshot renders correctly when there are no props 1`] = `
<div
  className="MuiPaper-root MuiCard-root makeStyles-root-1  MuiPaper-elevation1 MuiPaper-rounded"
  data-testid="card"
  style={
    Object {
      "background": "primary",
    }
  }
>
  <div
    className="makeStyles-targetInfo-2"
  >
    <div
      className="makeStyles-targetName-3"
    >
      <h4
        className="MuiTypography-root MuiTypography-body1"
        data-testid="title"
      >
        
      </h4>
    </div>
    <div
      className="makeStyles-targetNumbers-4"
    >
      <h3
        className="MuiTypography-root MuiTypography-body1"
        data-testid="done"
      >
        0
      </h3>
      <h6
        className="MuiTypography-root MuiTypography-body1"
        data-testid="total"
      >
        /
        0
      </h6>
    </div>
  </div>
  <div
    className="makeStyles-targetMeta-5"
  >
    <div
      className="makeStyles-targetIcon-6"
    >
      <span
        aria-hidden={true}
        className="material-icons MuiIcon-root"
        data-testid="el-icon"
      >
        trending_up
      </span>
    </div>
    <div
      className="makeStyles-targetDone-7"
    >
      <p
        className="MuiTypography-root MuiTypography-body1"
        data-testid="weekly-stats"
      >
        1/1 this week
      </p>
    </div>
  </div>
</div>
`;

exports[`<TargetItem> snapshot renders correctly when there are title and stats 1`] = `
<div
  className="MuiPaper-root MuiCard-root makeStyles-root-1  MuiPaper-elevation1 MuiPaper-rounded"
  data-testid="card"
  style={
    Object {
      "background": "primary",
    }
  }
>
  <div
    className="makeStyles-targetInfo-2"
  >
    <div
      className="makeStyles-targetName-3"
    >
      <h4
        className="MuiTypography-root MuiTypography-body1"
        data-testid="title"
      >
        Pedicure
      </h4>
    </div>
    <div
      className="makeStyles-targetNumbers-4"
    >
      <h3
        className="MuiTypography-root MuiTypography-body1"
        data-testid="done"
      >
        1
      </h3>
      <h6
        className="MuiTypography-root MuiTypography-body1"
        data-testid="total"
      >
        /
        4
      </h6>
    </div>
  </div>
  <div
    className="makeStyles-targetMeta-5"
  >
    <div
      className="makeStyles-targetIcon-6"
    >
      <span
        aria-hidden={true}
        className="material-icons MuiIcon-root"
        data-testid="el-icon"
      >
        trending_up
      </span>
    </div>
    <div
      className="makeStyles-targetDone-7"
    >
      <p
        className="MuiTypography-root MuiTypography-body1"
        data-testid="weekly-stats"
      >
        1/1 this week
      </p>
    </div>
  </div>
</div>
`;

exports[`<TargetItem> snapshot renders correctly when there stats are passed 1`] = `
<div
  className="MuiPaper-root MuiCard-root makeStyles-root-1  MuiPaper-elevation1 MuiPaper-rounded"
  data-testid="card"
  style={
    Object {
      "background": "primary",
    }
  }
>
  <div
    className="makeStyles-targetInfo-2"
  >
    <div
      className="makeStyles-targetName-3"
    >
      <h4
        className="MuiTypography-root MuiTypography-body1"
        data-testid="title"
      >
        
      </h4>
    </div>
    <div
      className="makeStyles-targetNumbers-4"
    >
      <h3
        className="MuiTypography-root MuiTypography-body1"
        data-testid="done"
      >
        0
      </h3>
      <h6
        className="MuiTypography-root MuiTypography-body1"
        data-testid="total"
      >
        /
        3
      </h6>
    </div>
  </div>
  <div
    className="makeStyles-targetMeta-5"
  >
    <div
      className="makeStyles-targetIcon-6"
    >
      <span
        aria-hidden={true}
        className="material-icons MuiIcon-root"
        data-testid="el-icon"
      >
        trending_up
      </span>
    </div>
    <div
      className="makeStyles-targetDone-7"
    >
      <p
        className="MuiTypography-root MuiTypography-body1"
        data-testid="weekly-stats"
      >
        1/1 this week
      </p>
    </div>
  </div>
</div>
`;

exports[`<TargetItem> snapshot renders correctly when there title is passed 1`] = `
<div
  className="MuiPaper-root MuiCard-root makeStyles-root-1  MuiPaper-elevation1 MuiPaper-rounded"
  data-testid="card"
  style={
    Object {
      "background": "primary",
    }
  }
>
  <div
    className="makeStyles-targetInfo-2"
  >
    <div
      className="makeStyles-targetName-3"
    >
      <h4
        className="MuiTypography-root MuiTypography-body1"
        data-testid="title"
      >
        Shave
      </h4>
    </div>
    <div
      className="makeStyles-targetNumbers-4"
    >
      <h3
        className="MuiTypography-root MuiTypography-body1"
        data-testid="done"
      >
        0
      </h3>
      <h6
        className="MuiTypography-root MuiTypography-body1"
        data-testid="total"
      >
        /
        0
      </h6>
    </div>
  </div>
  <div
    className="makeStyles-targetMeta-5"
  >
    <div
      className="makeStyles-targetIcon-6"
    >
      <span
        aria-hidden={true}
        className="material-icons MuiIcon-root"
        data-testid="el-icon"
      >
        trending_up
      </span>
    </div>
    <div
      className="makeStyles-targetDone-7"
    >
      <p
        className="MuiTypography-root MuiTypography-body1"
        data-testid="weekly-stats"
      >
        1/1 this week
      </p>
    </div>
  </div>
</div>
`;
